airbyte-local-cli-nodejs/src/command.ts

```diff
@@ -0,0 +1,286 @@
+import {Command, Option} from 'commander';
+
+import {logger, parseConfigFile, updateLogLevel} from './utils';
+import {CLI_VERSION} from './version';
+
+// Cli options provided by the user
+interface CliOptions {
+  // source and destination config
+  configFile?: string;
+  wizard?: object;
+
+  // source
+  srcImage: string;
+  srcConfig?: Record<string, any>;
+  srcOutputFile?: string;
+  srcCheckConnection?: boolean;
+  srcOnly?: boolean;
+  srcPull?: boolean;
+
+  // destination
+  dstImage: string;
+  dstConfig?: Record<string, any>;
+  dstUseHostNetwork?: boolean;
+  dstOnly?: string;
+  dstPull?: boolean;
+
+  // general connector settings
+  connectionName?: string;
+  stateFile: string;
+  fullRefresh?: boolean;
+  rawMessages?: boolean;
+  keepContainers?: boolean;
+  logLevel?: string;
+
+  // logging
+  debug?: boolean;
+}
+
+// Airbyte connector source or destination config
+export interface AirbtyeConfig {
+  image: string;
+  config?: object;
+  catalog?: object;
+  dockerOptions?: string;
+}
+
+enum AirbyteConfigInputType {
+  FILE = 'file',
+  OPTION = 'option',
+}
+
+// Config that are needed for running this Airbyte Cli
+export interface FarosConfig {
+  src?: AirbtyeConfig;
+  dst?: AirbtyeConfig;
+  srcOutputFile: string | undefined; // if srcOnly is true
```

Comment:
Can do `<name>?: <type>` instead of `<name>: <type> | undefined` for all these properties just like you did for src and dst

Reply:
my understanding is that as the optional fields are from CliOptions, they turned to `string | undefined` in FarosConfig. 
<img width="740" alt="Screenshot 2024-11-13 at 10 08 57" src="https://github.com/user-attachments/assets/bbd5e84d-5a13-4636-942b-1bc4634c8467">
it seems like I will have to do additional checking to avoid the type complaining which i think it's probably not worth it.... :))))) correct me if i'm wrong 

I gave all the boolean type fields default values, which i think probably is a better idea anyway. 
but keep the string type fields the same.